subsys/fs/shell.c

```diff
@@ -200,7 +201,10 @@ static int cmd_ls(const struct shell *sh, size_t argc, char **argv)
 			break;
 		}
 
-		shell_print(sh, "%s%s", entry.name, (entry.type == FS_DIR_ENTRY_DIR) ? "/" : "");
+		name_end = (entry.type == FS_DIR_ENTRY_DIR) ? "/" : "";
+		COND_CODE_1(CONFIG_FILE_SYSTEM_SHELL_LS_SIZE,
+			(shell_print(sh, "%8zu %s%s", entry.size, entry.name, name_end)),
+			(shell_print(sh, "%s%s", entry.name, name_end)));
```

Comment:
Prefer runtime checks over macro expansions. (compiler can optimize it already)

```suggestion
		if (IS_ENABLED(CONFIG_FILE_SYSTEM_SHELL_LS_SIZE)) {
			shell_print(sh, "%8zu %s%s", entry.size, entry.name, name_end);
		} else {
			shell_print(sh, "%s%s", entry.name, name_end);
		}
```